crates/services/tx_status_manager/src/manager.rs

```diff
@@ -745,6 +584,64 @@ mod tests {
             .collect()
     }
 
+    #[tokio::main(start_paused = true, flavor = "current_thread")]
+    #[allow(clippy::arithmetic_side_effects)]
+    async fn _test_status_manager_pruning(ttl: Duration, actions: Vec<Action>) {
+        let tx_status_change = TxStatusChange::new(100, Duration::from_secs(360));
+
+        let mut tx_status_manager = TxStatusManager::new(tx_status_change, ttl);
+
+        let tx_id_pool = generate_tx_id_pool();
+
+        // This will be used to track when each txid was updated so that
+        // we can do the final assert against the TTL.
+        let mut update_times = HashMap::new();
+
+        // Simulate flow of time and transaction updates
+        for action in actions {
+            match action {
+                Action::UpdateStatus { tx_id_index } => {
+                    let tx_id = tx_id_pool[tx_id_index];
+                    tx_status_manager.status_update(tx_id.into(), status_1());
+                    update_times.insert(tx_id, Instant::now());
+                }
+                Action::AdvanceTime { seconds } => {
+                    tokio::time::advance(Duration::from_secs(seconds)).await;
+                }
+            }
+        }
+
+        // Trigger the final pruning, making sure we use ID that is not
+        // in the pool
+        let final_tx_id = {
+            let marker: u64 = 0xDEADBEEF;
+            let mut id = [0u8; 32];
+            id[0..8].copy_from_slice(&marker.to_le_bytes());
+            id
+        };
+        assert!(!tx_id_pool.contains(&final_tx_id));
+        tx_status_manager.status_update(final_tx_id.into(), status_1());
```

Comment:
Maybe it's good to also have the last transaction in the `update_times` collection.

```suggestion
        tx_status_manager.status_update(final_tx_id.into(), status_1());
        update_times.insert(final_tx_id, Instant::now());
```